### PR TITLE
Thoth Thamos CLI

### PR DESCRIPTION
**Pull request by saisankargochhayat**
This tutorial demos using Thoth's advise using Thamos CLI
Link to the tutorial - [link](https://www.katacoda.com/saisankargochhayat/scenarios/thamos-cli)

Please review cc- @4n4nd @fridex @hemajv

### CODE REVIEW

File: ai-machine-learning/thoth-thamos/advise.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Thamos CLI runs based on the `.thoth.yaml` config, which we ahve setup already in the example repository here.
```
Comment (fridex) [inline ai-machine-learning/thoth-thamos/advise.md]: s/ahve/have/

File: ai-machine-learning/thoth-thamos/advise.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+Thamos CLI runs based on the `.thoth.yaml` config, which we ahve setup already in the example repository here. 
+
+You can browse the same in the editor or via terminal using -
```
Comment (fridex) [inline ai-machine-learning/thoth-thamos/advise.md]: `You can browse the config file` ?

File: ai-machine-learning/thoth-thamos/advise.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+``cat requirements.txt``{{execute}}
+
+You can check this out in the editor also. 
+
+Now lets try running the application after installing the depedendencies -
+
+``python3 -m venv venv``{{execute}}
+
+Then we jump into this environment -
```
Comment (fridex) [inline ai-machine-learning/thoth-thamos/advise.md]: Then we activate this virtual environment:

File: ai-machine-learning/thoth-thamos/intro.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+In this tutorial, we are going to run Thamos Advise on a simple Python app using Thamos CLI.
```
Comment (fridex) [inline ai-machine-learning/thoth-thamos/intro.md]: In this tutorial, we are going to run Thamos Advise on a simple TensorFlow Python app using Thamos CLI.

File: ai-machine-learning/thoth-thamos/setup.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+Before start using the Thamos let's setup the environment.
+
+This image comes with Python2 so we are going to install Python3 -
+
+``yum install python3``{{execute}}
```
Comment (fridex) [inline ai-machine-learning/thoth-thamos/setup.md]: Is there a need to install `python3-pip`?

**Review by fridex — COMMENTED** [review]
Very nice!

I wonder about the click error. When I was trying the scenario from the Katacoda environment with OpenShift, it looked like it works.

Nice job! :+1:

File: ai-machine-learning/thoth-thamos/setup.md (hunk 5 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/setup.md]: It seems to work fine without doing that.

Comment (fridex) [reply ai-machine-learning/thoth-thamos/setup.md]: OK, thanks for checking it? :+1:

File: ai-machine-learning/thoth-thamos/advise.md (hunk 1 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/advise.md]: thanks

**Comment by saisankargochhayat** [comment]
> Very nice!
> 
> I wonder about the click error. When I was trying the scenario from the Katacoda environment with OpenShift, it looked like it works.
> 
> Nice job!

@fridex Thanks, I have resolved those suggestions :smile:

File: ai-machine-learning/thoth-thamos/setup.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+Now that all that is done, let's check what Thamos offer-
+
+``thamos --help``{{execute}}
+
+You should see something like this -
+![Thamos help](https://raw.githubusercontent.com/saisankargochhayat/katacoda-scenarios/master/thamos-cli/assets/thamos_help.png)
+
+Bravo if you made it till here! We are almost done with the setup. 
+Last step is cloning out demo app - [Link](https://github.com/thoth-station/s2i-example)
```
Comment (4n4nd) [inline ai-machine-learning/thoth-thamos/setup.md]: is it supposed to be `our demo app` maybe?

File: ai-machine-learning/thoth-thamos/advise.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
+Thamos CLI runs based on the `.thoth.yaml` config, which we have setup already in the example repository here. 
+
+You can browse the config file in the editor or via terminal using -
+
+``cat .thoth.yaml``{{execute}}
```
Comment (4n4nd) [inline ai-machine-learning/thoth-thamos/advise.md:5]: I think you can do something like `filename.yaml`{{open}}, and it should create a hyperlink that directly opens the file in the editor. Just a suggestion might make it look cleaner

File: ai-machine-learning/thoth-thamos/advise.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+``source venv/bin/activate``{{execute}}
+
+And install the depedencies then - 
+
+``pip3 install -r requirements.txt``{{execute}}
+
+And finally, run the application (the virtual environment needs to be still activated):
+
+``python3 ./app.py``{{execute}}
```
Comment (4n4nd) [inline ai-machine-learning/thoth-thamos/advise.md:42]: When I tried to run this command in my katacoda env, I got this output:
Screenshot from 2020-04-07 00-32-19

File: ai-machine-learning/thoth-thamos/advise.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+``cat .thoth.yaml``{{execute}} 
+
+Now we are ready to do ask Thoth for Advise -
+
+``thamos advise``{{execute}}
+
+Tip - ``thamos advise --force`` runs the CLI without cache.
+
+Be a little patient here, this takes some time. After this is done you should see -
```
Comment (4n4nd) [inline ai-machine-learning/thoth-thamos/advise.md]: Can you add a `<br>` at the end of this line, so the image always shows below the text?
Same for any other images

**Review by 4n4nd — CHANGES_REQUESTED** [review]
A few minor changes, other than that LGTM :+1:

File: ai-machine-learning/thoth-thamos/advise.md (hunk 8 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/advise.md:42]: I think that is the expected output.
@fridex

File: ai-machine-learning/thoth-thamos/advise.md (hunk 7 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/advise.md:5]: Somehow it seems to only work on root directory. Like doesn't look for the file in subfolders. Do you have to put a full path link?

File: ai-machine-learning/thoth-thamos/setup.md (hunk 6 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/setup.md]: thanks

File: ai-machine-learning/thoth-thamos/advise.md (hunk 7 above)
Comment (4n4nd) [reply ai-machine-learning/thoth-thamos/advise.md:5]: I don't know, could you test it and lmk?

File: ai-machine-learning/thoth-thamos/advise.md (hunk 8 above)
Comment (fridex) [reply ai-machine-learning/thoth-thamos/advise.md:42]: Yes, that's expected output :+1: 

To remove warnings we can either adjust numpy or disable warnings in the code as discussed in https://github.com/tensorflow/tensorflow/issues/30427

To remove the instruction warning, we can possibly show it in the next step/scenario where thoth recommends optimized AICoE wheel files.

Comment (4n4nd) [reply ai-machine-learning/thoth-thamos/advise.md:42]: as long it is intentional :+1:

File: ai-machine-learning/thoth-thamos/advise.md (hunk 7 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/advise.md:5]: @4n4nd I have updated the file links :link:

File: ai-machine-learning/thoth-thamos/advise.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+Thamos CLI runs based on the `.thoth.yaml` config, which we have setup already in the example repository here. 
+
+You can browse the config file in the editor or via terminal using -
+
+``cat .thoth.yaml``{{execute}} 
+
+Now we are ready to do ask Thoth for Advise -
+
+``thamos advise``{{execute}}
```
Comment (hemajv) [inline ai-machine-learning/thoth-thamos/advise.md:9]: When I tried to run this in my Katacoda env, I got this as my output:
Screenshot from 2020-04-07 15-07-56

File: ai-machine-learning/thoth-thamos/intro.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+In this tutorial, we are going to run Thamos Advise on a simple TensorFlow Python app using Thamos CLI.
+Thus getting intelligent recommendation on the software stack during using the CLI.
```
Comment (hemajv) [inline ai-machine-learning/thoth-thamos/intro.md]: remove `during`?

File: ai-machine-learning/thoth-thamos/setup.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
+Before start using the Thamos let's setup the environment.
```
Comment (hemajv) [inline ai-machine-learning/thoth-thamos/setup.md]: change to: "Before we start using Thamos, let's give a couple of minutes for the environment to setup."?

File: ai-machine-learning/thoth-thamos/advise.md (hunk 10 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/advise.md:9]: Sorry about that, we are facing some issues with our prod env, should be fixed soon.

File: ai-machine-learning/thoth-thamos/setup.md (hunk 12 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/setup.md]: Sounds good!

File: ai-machine-learning/thoth-thamos/intro.md (hunk 11 above)
Comment (saisankargochhayat) [reply ai-machine-learning/thoth-thamos/intro.md]: thanks

File: ai-machine-learning/thoth-thamos/advise.md (hunk 10 above)
Comment (4n4nd) [reply ai-machine-learning/thoth-thamos/advise.md:9]: Hey @saisankargochhayat any updates on this?

I had a server timeout error when I tried to run this scenario
Screenshot from 2020-04-14 11-01-15

Comment (harshad16) [reply ai-machine-learning/thoth-thamos/advise.md:9]: @4n4nd  can you test that again, it been fixed.

Comment (hemajv) [reply ai-machine-learning/thoth-thamos/advise.md:9]: works now :tada: 

Screenshot from 2020-04-14 14-28-36

**Comment by saisankargochhayat** [comment]
@4n4nd can we merge this?

**Comment by 4n4nd** [comment]
> @4n4nd can we merge this?

@saisankargochhayat I think @hemajv and @harshad16 were looking into it.

**Comment by 4n4nd** [comment]
let's merge this, we are just waiting on some backend stuff that should be done soon.
Thanks to @harshad16

**Review by 4n4nd — APPROVED** [review]
:+1: LGTM

**Comment by harshad16** [comment]
Thanks @4n4nd .

**Pull request merged**